tests/test_std.rs

```diff
@@ -986,6 +986,32 @@ fn permutations_zero() {
     it::assert_equal((0..0).permutations(0), vec![vec![]]);
 }
 
+#[test]
+fn permutations_range_count() {
+    for n in 0..=7 {
+        for k in 0..=7 {
+            let len = if k <= n {
+                (n - k + 1..=n).product()
+            } else {
+                0
+            };
+            let mut it = (0..n).permutations(k);
+            assert_eq!(len, it.clone().count());
+            assert_eq!(len, it.size_hint().0);
+            assert_eq!(Some(len), it.size_hint().1);
+            for count in (0..len).rev() {
+                let elem = it.next();
+                assert!(elem.is_some());
+                assert_eq!(count, it.clone().count());
+                assert_eq!(count, it.size_hint().0);
+                assert_eq!(Some(count), it.size_hint().1);
+            }
+            let should_be_none = it.next();
+            assert!(should_be_none.is_none());
+        }
+    }
+}
```

Comment:
Is it possible to test an overflowing case by taking permutations from an `usize::MAX` stream of `()`? If such a test would take prohibitively long to run, we don't need it.

Reply:
I'm not sure to understand "_an `usize::MAX` stream of `()`_". Could you be more specific (a small code snippet maybe) about what I should test (assuming it's fast enough)?

Reply:
Maybe `std::iter::repeat(()).combinations(2)`.

Reply:
Maybe `std::iter::repeat(()).combinations(2)`.

Reply:
Done.

Reply:
Yes, this is great! Thank you!